code/menuui/mainhallmenu.cpp

```diff
@@ -2759,76 +2810,116 @@ void main_hall_reload_door(int region)
 }
 
 /**
- * Make the vasudan main hall funny
+ * Toggle the selected door and background between the provided headz animation and the original
  */
-void main_hall_vasudan_funny()
+void main_hall_set_door_headz(bool init)
 {
-	if (!Main_hall || !main_hall_is_vasudan())
-		return;
+	static bool Enable_headz = false;
+	static SCP_string last_main_hall_name;
+	
+	// Only toggle if we're not initializing the mainhall ui
+	// For our purposes here, "initializing" is whenever the mainhall game state is started from any other game state
+	// Since we're using static now, this kludge makes sure we use the last headz state
+	if (!init) {
+		Enable_headz = !Enable_headz;
+	} else {
+		// if we are initializing and the mainhall has changed then set everything back to default
+		if (Main_hall->name != last_main_hall_name) {
+			Enable_headz = false;
+			return;
+		}
 
-	static bool Vasudan_funny = false;
-	static SCP_string serious_bitmap;
-	static SCP_vector<std::tuple<int, SCP_string, interface_snd_id, interface_snd_id>> serious_door_info;
+		// if we are initializing and Enable_headz is false, then we have nothing else to do here
+		if (!Enable_headz) {
+			return;
+		}
+	}
 
-	Vasudan_funny = !Vasudan_funny;
+	int region_index = -1;
 
-	if (Vasudan_funny) {
-		// save stuff so we can restore it later
-		serious_bitmap = Main_hall->bitmap;
-		serious_door_info.clear();
-		serious_door_info.emplace_back(OPTIONS_REGION, Main_hall->door_anim_name.at(OPTIONS_REGION), Main_hall->door_sounds.at(OPTIONS_REGION).first, Main_hall->door_sounds.at(OPTIONS_REGION).second);
+	// Add some kludge to get the region from the mainhall and check that we have everything we need
+	if ((Main_hall->headz_index >= 0) && (Main_hall->headz_index < Main_hall->num_door_animations) && !Main_hall->headz_anim.empty()) {
+		region_index = Main_hall->headz_index;
+	}
 
-		if (!stricmp(Main_hall->bitmap.c_str(), "vhall")) {
-			Main_hall->door_sounds.at(OPTIONS_REGION).first = InterfaceSounds::VASUDAN_BUP;
-			Main_hall->door_sounds.at(OPTIONS_REGION).second = InterfaceSounds::VASUDAN_BUP;
+	static SCP_string original_bitmap;
+	// Why is this a vector? There can only ever be one
```

Comment:
Honestly, after looking at everything, it does not look like it has to be a vector.  But if someone wanted to upgrade it later, it would be better to leave it as is.